common/djangoapps/student/models.py

```diff
@@ -1529,7 +1529,7 @@ def generate_enrollment_status_hash(cls, user):
 
         if not status_hash:
             enrollments = cls.enrollments_for_user(user).values_list('course_id', 'mode')
-            enrollments = [(e[0].lower(), e[1].lower()) for e in enrollments]
+            enrollments = [(str(e[0]).lower(), e[1].lower()) for e in enrollments]
```

Comment:
Did we come to a consensus whether we were using str or unicode (or six.text_type) for course keys?

Reply:
I think the current state of the conversation can be simplified to:

* unicode() is what is more common in our codebase currently.
* using str instead of unicode for python 3 compatibility is potentially out of scope.
* if a course id contains a non-ascii character, using unicode() here will break format() below, but we don't support course ids with non-ascii characters anyway.

So last I remember, the consensus was to begrudgingly use unicode().

Reply:
If we're going to use unicode() we may as well use six.text_type when we remember it and get the Py3 support.

Reply:
FWIW I used six.text_type.